devito/types/dimension.py

```diff
@@ -1060,6 +1061,34 @@ def __new__(cls, d, name):
     _pickle_kwargs = []
 
 
+class ThreadDimension(BasicDimension):
```

Comment:
Not sure about this one, why isn't it just a `DefaultDimension` or part of `Thread`

Reply:
what is `Thread` ?

I need a special Dimension because I need to be aware of the concept of "thread dimension", such that the declarator can place an `omp parallel region` around some mallocs (see `definitions.py`).

Reply:
Not sure what you mean by "special dimension", this is just a generic fixed size dimension (so a DefaultDimension) that can be attached to one of the `NThreads` classes or something.

Reply:
I need `is_Thread`